copy/single.go

```diff
@@ -50,13 +58,13 @@ type copySingleImageResult struct {
 
 // copySingleImage copies a single (non-manifest-list) image unparsedImage, using c.policyContext to validate
 // source image admissibility.
-func (c *copier) copySingleImage(ctx context.Context, unparsedImage *image.UnparsedImage, targetInstance *digest.Digest, requireCompressionFormatMatch bool) (copySingleImageResult, error) {
+func (c *copier) copySingleImage(ctx context.Context, opt copySingleImageOptions) (copySingleImageResult, error) {
```

Comment:
(Absolutely non-blocking, this works fine: personally I’m inclined to say that the critically-important parameters which are “natural” to the operation can remain as top-level, and only the a bit unusual ones need to be named in a struct. ”copy(this image, and here are some details for that)”.

OTOH if we continued to pass `unparsedImage` and `targetInstance` directly, that would probably leave us with a single-field struct right now, which is… mostly an investment for the future.